src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -231,6 +231,175 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::CreateNew
+    }
+    if (source.HasAntialiasingMode() || source.AntialiasingModeOverrideSource() != nullptr)
+    {
+        duplicated->AntialiasingMode(source.AntialiasingMode());
+    }
+    if (source.HasRetroTerminalEffect() || source.RetroTerminalEffectOverrideSource() != nullptr)
+    {
+        duplicated->RetroTerminalEffect(source.RetroTerminalEffect());
+    }
+    if (source.HasForceFullRepaintRendering() || source.ForceFullRepaintRenderingOverrideSource() != nullptr)
+    {
+        duplicated->ForceFullRepaintRendering(source.ForceFullRepaintRendering());
+    }
+    if (source.HasSoftwareRendering() || source.SoftwareRenderingOverrideSource() != nullptr)
+    {
+        duplicated->SoftwareRendering(source.SoftwareRendering());
+    }
+    if (source.HasColorSchemeName() || source.ColorSchemeNameOverrideSource() != nullptr)
+    {
+        duplicated->ColorSchemeName(source.ColorSchemeName());
+    }
+    if (source.HasForeground() || source.ForegroundOverrideSource() != nullptr)
+    {
+        duplicated->Foreground(source.Foreground());
+    }
+    if (source.HasBackground() || source.BackgroundOverrideSource() != nullptr)
+    {
+        duplicated->Background(source.Background());
+    }
+    if (source.HasSelectionBackground() || source.SelectionBackgroundOverrideSource() != nullptr)
+    {
+        duplicated->SelectionBackground(source.SelectionBackground());
+    }
+    if (source.HasCursorColor() || source.CursorColorOverrideSource() != nullptr)
+    {
+        duplicated->CursorColor(source.CursorColor());
+    }
+    if (source.HasHistorySize() || source.HistorySizeOverrideSource() != nullptr)
+    {
+        duplicated->HistorySize(source.HistorySize());
+    }
+    if (source.HasSnapOnInput() || source.SnapOnInputOverrideSource() != nullptr)
+    {
+        duplicated->SnapOnInput(source.SnapOnInput());
+    }
+    if (source.HasAltGrAliasing() || source.AltGrAliasingOverrideSource() != nullptr)
+    {
+        duplicated->AltGrAliasing(source.AltGrAliasing());
+    }
+    if (source.HasCursorShape() || source.CursorShapeOverrideSource() != nullptr)
+    {
+        duplicated->CursorShape(source.CursorShape());
+    }
+    if (source.HasCursorHeight() || source.CursorHeightOverrideSource() != nullptr)
+    {
+        duplicated->CursorHeight(source.CursorHeight());
+    }
+    if (source.HasBellStyle() || source.BellStyleOverrideSource() != nullptr)
+    {
+        duplicated->BellStyle(source.BellStyle());
+    }
+    if (source.HasPixelShaderPath() || source.PixelShaderPathOverrideSource() != nullptr)
+    {
+        duplicated->PixelShaderPath(source.PixelShaderPath());
+    }
+    if (source.HasBackgroundImageAlignment() || source.BackgroundImageAlignmentOverrideSource() != nullptr)
+    {
+        duplicated->BackgroundImageAlignment(source.BackgroundImageAlignment());
+    }
+    if (source.HasConnectionType())
+    {
+        duplicated->ConnectionType(source.ConnectionType());
+    }
+
+    return *duplicated;

```

Comment:
Just so we're clear: when you duplicate Command Prompt, what do you get? When you duplicate WSL, what do you get? When you have a `defaults` layer, and then duplicate a leaf profile, what do you get?

I think that Command Prompt will give you a full JSON block. I think that WSL will give you a partial JSON block. I think that having a `defaults` layer will _copy all of the defaults into the new json block_.

Reply:
This does not seem right to me?

Reply:
Solved the `defaults` layer issue by adding a `UserDefaults` origin tag and checking for it!